### PR TITLE
refactor(misconf): Update refactored dependencies

### PR DESCRIPTION
**Pull request by simar7**
## Description
This PR updates the defsec dependency along with the refactored subcomponents of it. 

## Related issues
- Close https://github.com/aquasecurity/trivy/issues/5138

## Related PRs (need to be merged first)
- [x] https://github.com/aquasecurity/defsec/pull/1460
- [x] https://github.com/aquasecurity/trivy-aws/pull/11
- [x] https://github.com/aquasecurity/trivy-iac/pull/14
- [x] https://github.com/aquasecurity/trivy-policies/pull/8
- [ ] https://github.com/aquasecurity/trivy/pull/5507

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [ ] I've added tests that prove my fix is effective or that my feature works.
- [ ] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -13,7 +13,7 @@ require (
 	github.com/NYTimes/gziphandler v1.1.1
 	github.com/alicebob/miniredis/v2 v2.30.4
 	github.com/aquasecurity/bolt-fixtures v0.0.0-20200903104109-d34e7f983986
-	github.com/aquasecurity/defsec v0.92.0
+	github.com/aquasecurity/defsec v0.92.1-0.20230925225724-4b598f8ffb36
```
Comment (simar7) [inline go.mod:16]: note: update this to latest defsec when https://github.com/aquasecurity/defsec/pull/1460 is merged

**Comment by knqyf263** [comment]
I think we should announce the builtin policies will be downloaded from `ghcr.io/aquasecurity/trivy-policies`.

**Comment by simar7** [comment]
> I think we should announce the built-in policies will be downloaded from `ghcr.io/aquasecurity/trivy-policies`. The `defsec` bundle will not be updated, right? If so, older Trivy versions won't follow the policy updates.

Good call. We do have `--policy-bundle-repository` for older users to use newer policies at the new location. We can include this information in the announcement.

**Comment by knqyf263** [comment]
>We do have --policy-bundle-repository for older users to use newer policies at the new location. We can include this information in the announcement.

Yeah, it's nice. Those who want to keep using older versions can specify `--policy-bundle-repository ghcr.io/aquasecurity/trivy-policies`.

**Comment by knqyf263** [comment]
@simar7 Can you please resolve conflicts?

**Pull request merged**